src/main/java/com/duke/util/Parser.java

```diff
@@ -0,0 +1,109 @@
+package com.duke.util;
+
+
+import com.duke.command.Command;
+import com.duke.command.DeleteCommand;
+import com.duke.command.DoneCommand;
+import com.duke.command.ExitCommand;
+import com.duke.command.FindCommand;
+import com.duke.command.ListCommand;
+import com.duke.command.TaskCommand;
+import com.duke.task.Deadline;
+import com.duke.task.Event;
+import com.duke.task.Task;
+import com.duke.task.Todo;
+
+import java.time.format.DateTimeParseException;
+import java.util.StringTokenizer;
+
+/**
+ * Represents a handler of the user input command which converts <code>String</code>
+ * commands into actual <code>Command</code> objects to be executed.
+ */
+public class Parser {
+    /**
+     * Takes in a <code>String</code> form of command and converts it into <code>Command</code>
+     * object to be executed.
+     *
+     * @param cmd String representation of user input
+     * @return Command representation of the user input
+     * @throws DukeException when the user input is invalid.
+     */
+    public static Command parse(String cmd) throws DukeException {
+        StringTokenizer st = new StringTokenizer(cmd);
+        String firstToken = st.nextToken();
+        Command output;
+
+        if (cmd.equals("bye")) {
+            output = new ExitCommand();
+        } else if (cmd.equals("list")) {
+            output = new ListCommand();
+        } else if (firstToken.equals("delete")) {
+            try {
+                int index = Integer.parseInt(cmd.substring(7)) - 1;
+                output = new DeleteCommand(index);
+            } catch (Exception e) {
```

Comment:
Exception caught is too general. Should try to catch more specific exception if possible.